### ISSUE TITLE
Reproducing GRPO based on Qwen2.5-1.5B-Instruct and using Math-220K dataset Yields Unexpected Results

### ISSUE THREAD

**Issue by undyingjoker** [huggingface/open-r1#538]
I am currently experimenting with the GPRO method in Open-R1, but the results have not met expectations.

Using Qwen2.5-1.5B-Instruct as the base models, Math-220k as the training dataset and MATH-500 as test sets.
However, as training progressed, the accuracy on the test sets steadily declined from 55.4 (Qwen2.5-1.5B-Instruct ) to 18.2 (Qwen2.5-1.5B-Instruct -GRPO)

```
ACCELERATE_LOG_LEVEL=info accelerate launch --config_file recipes/accelerate_configs/zero2.yaml \
    --num_processes=16 src/open_r1/grpo.py \
    --config recipes/Qwen2.5-1.5B-Instruct/grpo/config_demo.yaml &>> train_qwen__grpo_220k_rl_step.log
```

```
# Model arguments
model_name_or_path: /mnt/workspace/common/models/Qwen2.5-1.5B-Instruct
model_revision: main
torch_dtype: bfloat16
attn_implementation: flash_attention_2

# Data training arguments
dataset_name: /mnt/workspace/common/datas/OpenR1-Math-220k
system_prompt: "You are a helpful AI Assistant that provides well-reasoned and detailed responses. You first think about the reasoning process as an internal monologue and then provide the user with the answer. Respond in the following format: <think>\n...\n</think>\n<answer>\n...\n</answer>"

# GRPO trainer config
bf16: true
use_vllm: false
vllm_device: auto
vllm_gpu_memory_utilization: 0.7
do_eval: false
gradient_accumulation_steps: 2
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
hub_model_id: Qwen2.5-1.5B-Open-R1-GRPO
hub_strategy: every_save
learning_rate: 2.0e-05
log_completions: true
log_level: info
logging_first_step: true
logging_steps: 1
logging_strategy: steps
lr_scheduler_type: cosine
max_prompt_length: 512
max_completion_length: 1024
max_steps: -1
num_generations: 8
num_train_epochs: 1
output_dir: data/Qwen2.5-1.5B-Open-R1-GRPO
overwrite_output_dir: true
per_device_eval_batch_size: 16
per_device_train_batch_size: 16
push_to_hub: false
report_to:
- tensorboard
reward_funcs:
- accuracy
- format
- tag_count
reward_weights:
- 1.0
- 1.0
- 1.0
save_strategy: "steps"
save_steps: 100
save_total_limit: 10000000000000
seed: 42
warmup_ratio: 0.1

```

**Comment by ahatamiz** [comment]
why your learning rate is `learning_rate: 2.0e-05` ? that's too high.

**Comment by HwangYej1** [comment]
Did you try to do SFT on the this instruct model?

**Comment by undyingjoker** [comment]
> Did you try to do SFT on the this instruct model?

I didn't try it. I trained with GRPO.

**Comment by anuj-fujitsu** [comment]
I tried the same experiment, and after 1000 iterations, I am getting "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!" as completion.

**Comment by kyleliang919** [comment]
`sbatch --nodes 2 slurm/train.slurm --task grpo --accelerator zero2 --model DeepSeek-R1-Distill-Qwen-1.5B --config demo --dp 8 --args "--run_name grpo_baseline"`

I used this command and it gets a worse model after GRPO
